app/components/sso-settings/index.js

```
@@ -0,0 +1,194 @@
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency-decorators';
+import ENV from 'irene/config/environment';
+import parseError from 'irene/utils/parse-error';
+import { tracked } from '@glimmer/tracking';
+
+export default class SsoSettingsComponent extends Component {
+  @service intl;
+  @service ajax;
+  @service store;
+  @service('notifications') notify;
+
+  @tracked showDeleteIdpMetadataConfirm = false;
+  @tracked spMetadata = null;
+  @tracked idpMetadata = null;
+  @tracked idpMetadataXml = null;
+  @tracked spConfig = 'manual';
+
+  spMetadataKeys = [
+    { labelKey: 'entityID', valueKey: 'entity_id' },
+    { labelKey: 'acsURL', valueKey: 'acs_url' },
+    { labelKey: 'nameIDFormat', valueKey: 'named_id_format' },
+  ];
+
+  tPleaseTryAgain = this.intl.t('pleaseTryAgain');
+  tSomethingWentWrongContactSupport = this.intl.t(
+    'somethingWentWrongContactSupport'
+  );
+  tRemovedSSOSuccessfully = this.intl.t('removedSSOSuccessfully');
+
+  constructor() {
+    super(...arguments);
+
+    this.SPMetadata.perform();
+    this.getIdPMetadata.perform();
+  }
+
+  get sso() {
+    return this.store.queryRecord('organization-sso', {});
+  }
+
+  // Switch SP config format
+  @action
+  handleSpConfigChange(event) {
+    this.spConfig = event.target.value;
+  }
+
+  get spConfigIsManual() {
+    return this.spConfig == 'manual';
+  }
+
+  get spConfigIsXml() {
+    return this.spConfig == 'xml';
+  }
+
+  // Fetch SP Metadata
+  @task({ restartable: true })
+  *SPMetadata() {
+    const spMetadata = yield this.ajax.request(ENV.endpoints.saml2SPMetadata);
+    this.spMetadata = spMetadata;
+  }
+
+  // Fetch IdP Metadata
+  @task({ restartable: true })
+  *getIdPMetadata() {
+    try {
+      const idpMetadata = yield this.store.queryRecord(
+        'saml2-idp-metadata',
+        {}
+      );
+
+      this.idpMetadata = idpMetadata;
+    } catch (error) {
```

An empty catch block?